File: sensor_app/app/src/main/java/com/ifsc/sensorapp/MainActivity.java
package com.ifsc.sensorapp;

import androidx.appcompat.app.AppCompatActivity;

import android.hardware.Sensor;
import android.hardware.SensorEvent;
import android.hardware.SensorEventListener;
import android.hardware.SensorManager;
import android.os.Bundle;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity implements SensorEventListener  {
    SensorManager sensorManager;
    Sensor sensor;
    TextView tvSensorLight;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        sensorManager = (SensorManager) getSystemService(getApplicationContext().SENSOR_SERVICE);
        sensor = sensorManager.getDefaultSensor(Sensor.TYPE_LIGHT);
        sensorManager.registerListener(this, sensor, SensorManager.SENSOR_DELAY_NORMAL);
        tvSensorLight = findViewById(R.id.tvSensorLight);
    }

    @Override
    public void onSensorChanged(SensorEvent event) {
        float currentValue = event.values[0];
        tvSensorLight.setText(Float.toString(currentValue));
    }

    @Override
    public void onAccuracyChanged(Sensor sensor, int accuracy) {

    }
}

File: sorteio_app/sorteio_app_native/app/src/main/java/com/ifsc/sorteioapp/MainActivity.kt
package com.ifsc.sorteioapp

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.EditText
import android.widget.TextView
import kotlin.math.absoluteValue
import kotlin.math.min
import kotlin.random.Random

class MainActivity : AppCompatActivity() {
    lateinit var minEditText: EditText;
    lateinit var maxEditText: EditText;
    lateinit var resultText: TextView;

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        minEditText = findViewById<EditText>(R.id.minValueEditText);
        maxEditText = findViewById<EditText>(R.id.maxValueEditText);
        resultText = findViewById<TextView>(R.id.resultado);
    }

    fun draftValue(buttonView: View) {
        var min = getIntValueFromEditText(minEditText);
        var max = getIntValueFromEditText(maxEditText);

        if (!isValidForDraft(min, max))
            return;

        var result = Random.nextInt(min!!, max!! + 1);
        resultText.setText(result.toString());
    }

    private fun getIntValueFromEditText(editText: EditText): Int? {
        var text = editText.text.toString();

        if (text.isEmpty())
            return null;
        return text.toInt();
    }

    private fun isValidForDraft(min: Int?, max: Int?): Boolean {
        if (min == null || max == null) {
            resultText.setText("Informe um valor!");
            return false;
        }

        if (max < min) {
            resultText.setText("Escolha um intervalo válido!");
            return false;
        }

        return true;
    }
}